### DISCUSSION TITLE
How to fix tabufline showing nvim-tree separator on the left, when nvim-tree is oped on the right?

### DISCUSSION THREAD

**Discussion by Andrew15-5** [NvChad/NvChad#2429] [Q&A] [answered]
I've seen a related question: #2329, but it seems that there was a bug of some sort. But I actually don't have a bug, just can't figure out how to update tabufline with the `nvim-tree` config:

```lua
view = {
  side = "right"
}
```

Since tabufline is a NvChad's plugin, all I have is https://nvchad.com/docs/config/nvchad_ui#override_tabufline_modules, which I think is not helpful at all. Can you show the magic of configuration?)

![image](https://github.com/NvChad/NvChad/assets/37143421/92b36b12-ddef-4ba0-b90a-20fa7a31bfd1)
![image](https://github.com/NvChad/NvChad/assets/37143421/1db3c059-0948-494e-a879-98bf77213dfa)

**Comment by siduck** [comment, marked as answer]
@Andrew15-5 please override tabufline module, to keep the separator on right

**Reply by Andrew15-5** [reply]
I don't know how! It's there a documentation for this module and how to override it?

**Reply by siduck** [reply]
@Andrew15-5 https://nvchad.com/docs/config/nvchad_ui

```lua
function(modules)
    table.insert(modules, modules[1])
   table.remove(modules[1])
end
```

Gave u a hint! once u learn the docs, you will know where to put this and understand the index stuff

**Reply by Andrew15-5** [reply]
The documentation doesn't say anything, it just says something about "modules" and a random index 4. When I opened a link at the bottom I finally found what are these "modules", thanks to the search on page function in Firefox:

https://github.com/NvChad/ui/blob/0d45d6053455b017f951251b693b6666975c0ebd/lua/nvchad/tabufline/modules.lua#L202-L207:

```lua
  local modules = {
    M.NvimTreeOverlay(),
    M.bufferlist(),
    M.tablist(),
    M.buttons(),
  }
```

And `table.remove(modules[1])` is not a valid Lua command. But now I understand the basic idea of what it all means and how to use it, thanks!

```lua
  tabufline = {
    enabled = true,
    overriden_modules = function(modules)
      table.insert(modules, modules[1])
      table.remove(modules, 1)
    end,
  },
```

I only wish that they (modules) have been named instead of indexed, feels less human-oriented.

**Reply by siduck** [reply]
@Andrew15-5  it wouldnt be easy to manipulate if they werent indexed, i'm glad you made it work. passed the test!
